internals/data_backup.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from googleapiclient.discovery import build
+from googleapiclient.discovery_cache.base import Cache
+
+from framework import basehandlers
+
+class MemoryCache(Cache):
+    _CACHE = {}
+
+    def get(self, url):
+        return MemoryCache._CACHE.get(url)
+
+    def set(self, url, content):
+        MemoryCache._CACHE[url] = content
+
+
+class BackupExportHandler(basehandlers.FlaskHandler):
+  """Triggers a new Datastore export."""
+
+  def get_template_data(self):
+    bucket = os.environ.get("BACKUP_BUCKET")
```

Comment:
What sets this environment variable?
If you want to use an environment variable, it could be set in app.yaml, but that file is the same for staging and prod.
However, it might be more clear to set it as a constant in settings.py.  There are conditional sections there for staging and prod.